npipe_windows.go

```diff
@@ -443,6 +446,12 @@ func (c *PipeConn) Write(b []byte) (int, error) {
 
 // Close closes the connection.
 func (c *PipeConn) Close() error {
+	c.closedMutex.Lock()
+	defer c.closedMutex.Unlock()
+	if c.closed {
+		return nil
+	}
+	c.closed = true
 	return syscall.CloseHandle(c.handle)
```

Comment:
Makes the code alot uglier to not do it this way.  I can't imagine any deadlock by having the second guy wait for the first to complete the system call.